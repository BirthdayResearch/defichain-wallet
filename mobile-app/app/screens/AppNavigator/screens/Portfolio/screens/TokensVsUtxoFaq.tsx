import { ThemedScrollViewV2, ThemedTextV2 } from "@components/themed";
import {
  WalletAccordionV2,
  AccordionContent,
} from "@components/WalletAccordionV2";
import { tailwind } from "@tailwind";
import { translate } from "@translations";

export function TokensVsUtxoFaq(): JSX.Element {
  const faqContent: AccordionContent[] = [
    {
      title: translate("components/UtxoVsTokenFaq", "What are UTXOs?"),
      content: [
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "DFI in UTXO form is the primary form of DFI. It is used for core cryptocurrency purposes such as send, receive and fees."
          ),
          type: "paragraph",
        },
      ],
    },
    {
      title: translate(
        "components/UtxoVsTokenFaq",
        "What are the use cases for UTXO?"
      ),
      content: [
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "Use cases for UTXO include (but not limited) to the following:"
          ),
          type: "paragraph",
        },
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "Sending DFI to other wallets"
          ),
          type: "bullet",
        },
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "Receiving DFI from other wallets"
          ),
          type: "bullet",
        },
        {
          text: translate("components/UtxoVsTokenFaq", "Transaction fees"),
          type: "bullet",
        },
      ],
    },
    {
      title: translate("components/UtxoVsTokenFaq", "What are Tokens?"),
      content: [
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "DFIs in token form are used for various DeFiChain functions such as Liquidity Pools, Loans, Auctions and DEX swaps."
          ),
          type: "paragraph",
        },
      ],
    },
    {
      title: translate(
        "components/UtxoVsTokenFaq",
        "What are the use cases for Tokens?"
      ),
      content: [
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "Use cases for Tokens include (but not limited) to the following:"
          ),
          type: "paragraph",
        },
        {
          text: translate(
            "components/UtxoVsTokenFaq",
            "Swapping DFI with other tokens (e.g. dBTC, dETH, dUSDT)"
          ),
          type: "bullet",
        },
        {
          text: translate("components/UtxoVsTokenFaq", "Liquidity pools"),
          type: "bullet",
        },
        {
          text: translate("components/UtxoVsTokenFaq", "Collateral for Vaults"),
          type: "bullet",
        },
        {
          text: translate("components/UtxoVsTokenFaq", "Auction bidding"),
          type: "bullet",
        },
      ],
    },
  ];

  return (
    <ThemedScrollViewV2
      contentContainerStyle={tailwind("pt-8 px-5 pb-16")}
      style={tailwind("flex-1")}
      testID="utxo_vs_token_faq"
    >
      <ThemedTextV2 style={tailwind("text-base font-normal-v2 px-5")}>
        {translate(
          "components/UtxoVsTokenFaq",
          "DFI exists in two forms: UTXO and Token. They can be converted with one another to serve different use cases within your light wallet."
        )}
      </ThemedTextV2>

      <WalletAccordionV2
        testID="loans_faq_accordion"
        activeSections={[0]}
        title={translate(
          "components/UtxoVsTokenFaq",
          "FREQUENTLY ASKED QUESTIONS"
        )}
        content={faqContent}
      />
    </ThemedScrollViewV2>
  );
}
