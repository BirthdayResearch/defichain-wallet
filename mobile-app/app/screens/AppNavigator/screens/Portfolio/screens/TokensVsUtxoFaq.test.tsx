import { render } from "@testing-library/react-native";
import { TokensVsUtxoFaq } from "./TokensVsUtxoFaq";

describe("Tokens vs Utxo FAQ screen", () => {
  it("should match snapshot", async () => {
    const rendered = render(<TokensVsUtxoFaq />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
