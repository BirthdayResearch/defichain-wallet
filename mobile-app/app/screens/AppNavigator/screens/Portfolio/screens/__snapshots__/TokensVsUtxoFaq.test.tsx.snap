// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens vs Utxo FAQ screen should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={{}}
  style={
    [
      {},
      {},
    ]
  }
  testID="utxo_vs_token_faq"
>
  <View>
    <Text
      style={
        [
          {},
          [
            {},
            {},
          ],
        ]
      }
    >
      DFI exists in two forms: UTXO and Token. They can be converted with one another to serve different use cases within your light wallet.
    </Text>
    <RCTScrollView
      style={
        [
          undefined,
          {},
        ]
      }
      testID="loans_faq_accordion"
    >
      <View>
        <Text
          style={
            [
              {},
              [
                {},
                {},
              ],
            ]
          }
          text="FREQUENTLY ASKED QUESTIONS"
        >
          FREQUENTLY ASKED QUESTIONS
        </Text>
        <View
          style={
            [
              {},
              {},
            ]
          }
        >
          <View
            style={{}}
          >
            <View
              accessibilityState={
                {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  [
                    [
                      {},
                      {},
                      false,
                    ],
                    {},
                  ]
                }
              >
                <Text
                  style={
                    [
                      {},
                      [
                        {},
                        {},
                      ],
                    ]
                  }
                >
                  What are UTXOs?
                </Text>
                <Text />
              </View>
            </View>
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={{}}
              >
                <View
                  style={
                    [
                      [
                        {},
                        false,
                      ],
                      {},
                    ]
                  }
                >
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      DFI in UTXO form is the primary form of DFI. It is used for core cryptocurrency purposes such as send, receive and fees.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={{}}
          >
            <View
              accessibilityState={
                {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  [
                    [
                      {},
                      {},
                      false,
                    ],
                    {},
                  ]
                }
              >
                <Text
                  style={
                    [
                      {},
                      [
                        {},
                        {},
                      ],
                    ]
                  }
                >
                  What are the use cases for UTXO?
                </Text>
                <Text />
              </View>
            </View>
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={{}}
              >
                <View
                  style={
                    [
                      [
                        {},
                        false,
                      ],
                      {},
                    ]
                  }
                >
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Use cases for UTXO include (but not limited) to the following:
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Sending DFI to other wallets
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Receiving DFI from other wallets
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Transaction fees
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={{}}
          >
            <View
              accessibilityState={
                {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  [
                    [
                      {},
                      {},
                      false,
                    ],
                    {},
                  ]
                }
              >
                <Text
                  style={
                    [
                      {},
                      [
                        {},
                        {},
                      ],
                    ]
                  }
                >
                  What are Tokens?
                </Text>
                <Text />
              </View>
            </View>
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={{}}
              >
                <View
                  style={
                    [
                      [
                        {},
                        false,
                      ],
                      {},
                    ]
                  }
                >
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      DFIs in token form are used for various DeFiChain functions such as Liquidity Pools, Loans, Auctions and DEX swaps.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={{}}
          >
            <View
              accessibilityState={
                {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  [
                    [
                      {},
                      false,
                      false,
                    ],
                    {},
                  ]
                }
              >
                <Text
                  style={
                    [
                      {},
                      [
                        {},
                        {},
                      ],
                    ]
                  }
                >
                  What are the use cases for Tokens?
                </Text>
                <Text />
              </View>
            </View>
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={{}}
              >
                <View
                  style={
                    [
                      [
                        {},
                        false,
                      ],
                      {},
                    ]
                  }
                >
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Use cases for Tokens include (but not limited) to the following:
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Swapping DFI with other tokens (e.g. dBTC, dETH, dUSDT)
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Liquidity pools
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Collateral for Vaults
                    </Text>
                  </View>
                  <View
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        [
                          {},
                          [
                            {},
                            {},
                          ],
                        ]
                      }
                    >
                      Auction bidding
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
